sys/fmt/fmt.c

```diff
@@ -400,6 +415,16 @@ size_t fmt_lpad(char *out, size_t in_len, size_t pad_len, char pad_char)
     return pad_len;
 }
 
+void fmt_to_lower(char *str)
```

Comment:
The rest of the `fmt_*` functions take the length if the string as a parameter. Doing that will allow you to handle embedded zeros.